### PR TITLE
Add controller for ElasticSearch admin

### PR DESCRIPTION
**Pull request by JohnKallies**
This still needs a documentation update in the READMEs. I don't se any means of testing this with existing testing strategies, but I would prefer to if reasonable.

### CODE REVIEW

File: api/src/main/kotlin/edu/wgu/osmt/collection/CollectionEsRepo.kt [hunk 1]
```diff
@@ -34,6 +36,10 @@ interface CustomCollectionQueries : FindsAllByPublishStatus<CollectionDoc> {
             Sort.by("name.keyword").descending()
         )
     ): SearchHits<CollectionDoc>
+
+    fun deleteIndex() {
+        elasticSearchTemplate.indexOps(IndexCoordinates.of(INDEX_COLLECTION_DOC)).delete();
```
Comment (jchavez137) [inline api/src/main/kotlin/edu/wgu/osmt/collection/CollectionEsRepo.kt:41]: we can skip semicolon here.

File: api/src/main/kotlin/edu/wgu/osmt/jobcode/JobCodeEsRepo.kt [hunk 2]
```diff
@@ -9,13 +10,18 @@ import org.springframework.beans.factory.annotation.Autowired
...
 import org.springframework.data.elasticsearch.core.query.NativeSearchQueryBuilder
 import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
 import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories
 
 interface CustomJobCodeRepository {
     val elasticSearchTemplate: ElasticsearchRestTemplate
     fun typeAheadSearch(query: String): SearchHits<JobCode>
+
+    fun deleteIndex() {
+        elasticSearchTemplate.indexOps(IndexCoordinates.of(INDEX_JOBCODE_DOC)).delete();
```
Comment (jchavez137) [inline api/src/main/kotlin/edu/wgu/osmt/jobcode/JobCodeEsRepo.kt:23]: semicolon..

File: api/src/main/kotlin/edu/wgu/osmt/keyword/KeywordEsRepo.kt [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
 import org.springframework.data.elasticsearch.core.query.NativeSearchQueryBuilder
 import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
 import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories
 
 interface CustomKeywordRepository {
     val elasticSearchTemplate: ElasticsearchRestTemplate
     fun typeAheadSearch(query: String, type: KeywordTypeEnum): SearchHits<Keyword>
+
+    fun deleteIndex() {
+        elasticSearchTemplate.indexOps(IndexCoordinates.of(INDEX_KEYWORD_DOC)).delete();
```
Comment (jchavez137) [inline api/src/main/kotlin/edu/wgu/osmt/keyword/KeywordEsRepo.kt:21]: semicolon

File: api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillEsRepo.kt [hunk 4]
```diff
@@ -46,6 +48,11 @@ interface CustomRichSkillQueries : FindsAllByPublishStatus<RichSkillDoc> {
     fun findSimilar(apiSimilaritySearch: ApiSimilaritySearch): SearchHits<RichSkillDoc>
 
     fun occupationQueries(query: String): NestedQueryBuilder
+
+
+    fun deleteIndex() {
+        elasticSearchTemplate.indexOps(IndexCoordinates.of(INDEX_RICHSKILL_DOC)).delete();
```
Comment (jchavez137) [inline api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillEsRepo.kt:54]: we can skip semicolon

File: api/src/main/kotlin/edu/wgu/osmt/collection/CollectionEsRepo.kt (hunk 1 above)
Comment (JohnKallies) [reply api/src/main/kotlin/edu/wgu/osmt/collection/CollectionEsRepo.kt:41]: Yup, good catch. I copy/pasted in from Java.

File: api/src/main/kotlin/edu/wgu/osmt/elasticsearch/ElasticSearchAdminController.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import edu.wgu.osmt.RoutePaths.ES_ADMIN_DELETE_INDICES
+import edu.wgu.osmt.RoutePaths.ES_ADMIN_REINDEX
+import edu.wgu.osmt.config.AppConfig
+import edu.wgu.osmt.security.OAuthHelper
+import edu.wgu.osmt.task.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.*
+import org.springframework.stereotype.Controller
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
```
[the remark this answers is not in the document]
Comment (JohnKallies) [reply api/src/main/kotlin/edu/wgu/osmt/elasticsearch/ElasticSearchAdminController.kt:12]: I thought I fixed my editor for that. Also a good catch.

**Review by jchavez137 — APPROVED** [review]
Looks good!

**Review by manuel-delvillar — APPROVED** [review]

**Pull request merged**